docs: clarify that Coder server should run on a separate machine (#27468)
coder/coder: docs/get-started/index.md

> [!TIP]
> If you use a coding agent like Claude Code, the [coder/skills](https://github.com/coder/skills) `setup` skill can train the coding agent on the following steps (install a container runtime, install Coder, create your first template, and launch a workspace).

> [!NOTE]
> This guide installs the Coder server on the same machine you're working from so you can try Coder quickly.

> > That's fine for this walkthrough, but Coder's real value comes from remote development: hosting the Coder server on a separate machine (a cloud VM, a spare desktop, or on-prem hardware) lets you and your team provision workspaces on infrastructure that's more powerful, always-on, and reachable from anywhere, instead of tying your dev environment to your own laptop.

Reviewer: ```suggestion
> That's fine for this tutorial, but Coder's real value comes from remote development.
> By hosting the Coder server on a separate machine (a cloud VM, a spare desktop, or on-premise hardware), you and your team can provision workspaces on infrastructure that's more powerful, always-on, and reachable from anywhere, instead of tying your dev environment to your own laptop.
```

1. Walkthrough has the word walk, but it may not be as inclusive a term as tutorial, so I prefer to use tutorial here.
2. Each sentence should be on its own line.

> > When you're ready to move past this quickstart, install the Coder server on a separate machine and connect to it remotely. Refer to [Installing Coder](../install/index.md) for supported platforms and installation methods.

Reviewer: ```suggestion
> When you're ready to move past this tutorial, install the Coder server on a separate machine and connect to it remotely. Refer to the [Install guide](../install/index.md) for supported platforms and installation methods.
```

No gerund (-ing verb) headings.

Install the `coder` CLI to get started:

> [!NOTE]
> For this quickstart, install and run the Coder server on the same machine you're using right now.

> > In a real deployment, install the Coder server on a separate machine (not your laptop) so its compute stays available and shareable independent of any one person's device.

Reviewer: ```suggestion
> In a real deployment, install the Coder server on a separate machine (not your laptop) so its compute stays available and shareable, independent of any one person's device.
```

> > Refer to [Installing Coder](../install/index.md) for details on installing on a separate host.

Reviewer: ```suggestion
> Refer to the [Install guide](../install/index.md) for details on installing on a separate host.
```
